api/v1beta1/azuremachine_default_test.go

```diff
@@ -453,9 +517,21 @@ func (m mockClient) Get(ctx context.Context, key client.ObjectKey, obj client.Ob
 	if m.ReturnError {
 		return errors.New("AzureCluster not found: failed to find owner cluster for test-cluster")
 	}
-	ac := &AzureCluster{}
-	ac.Spec.SubscriptionID = "test-subscription-id"
-	obj.(*AzureCluster).Spec.SubscriptionID = ac.Spec.SubscriptionID
+	// Check if we're calling Get on an AzureCluster or a Cluster
```

Comment:
This logic can probably be simplified with `switch obj.(type)` and `case *AzureCluster:`, `case *clusterv1.Cluster`, etc.

Reply:
Ah I wasn't aware of this method!! Thanks for the tip